lib/upnp/upnp.go

```diff
@@ -249,16 +276,39 @@ func parseResponse(ctx context.Context, deviceType string, resp []byte) ([]IGDSe
 	}
 
 	deviceDescriptionURL, err := url.Parse(deviceDescriptionLocation)
-
 	if err != nil {
 		l.Infoln("Invalid IGD location: " + err.Error())
+		return nil, err
+	}
+
+	if err != nil {
+		l.Infoln("Invalid source IP for IGD: " + err.Error())
+		return nil, err
 	}
 
 	deviceUSN := response.Header.Get("USN")
 	if deviceUSN == "" {
 		return nil, errors.New("invalid IGD response: USN not specified")
 	}
 
+	deviceIP := net.ParseIP(deviceDescriptionURL.Hostname())
```

Comment:
According to the spec of the location header for UPnP:

> Required. Contains a URL to the UPnP description of the root device. **Normally the host portion contains a literal IP
address rather than a domain name** in unmanaged networks. Specified by UPnP vendor. Single URL.

There's a chance that we might get the local DNS name of the router.

Reply:
You're right, and localIP is actually used later. I've fixed that usage now (although the link-local replacement code isn't used if the address doesn't parse as an IPv6, so it shouldn't cause trouble there)